lib/SPIRV/SPIRVToLLVMDbgTran.cpp

```diff
@@ -450,6 +450,49 @@ SPIRVToLLVMDbgTran::transTypeSubrange(const SPIRVExtInst *DebugInst) {
                                      TranslatedOps[2], TranslatedOps[3]);
 }
 
+DIStringType *
+SPIRVToLLVMDbgTran::transTypeString(const SPIRVExtInst *DebugInst) {
+  using namespace SPIRVDebug::Operand::TypeString;
+  const SPIRVWordVec &Ops = DebugInst->getArguments();
+  assert(Ops.size() >= MinOperandCount && "Invalid number of operands");
+
+  StringRef Name = getString(Ops[NameIdx]);
+  DIType *BaseTy =
+      transDebugInst<DIType>(BM->get<SPIRVExtInst>(Ops[BaseTypeIdx]));
```

Comment:
Maybe we need to rethink the specification here. As you can see, clang-tidy failed because `BaseTy` is never used. It's because there are no possibility (on LLVM core lib side) to create type string with the encoding set.

Reply:
We shouldn't adjust SPIR-V spec with capabilities of `DIBuilder` class of LLVM. If there is no overload for [createStringType](https://llvm.org/doxygen/classllvm_1_1DIBuilder.html#a5153d3ea2754b6b7fa554daecdf4ea5d) it doesn't mean, that we should remove basic type (or encoding) from SPIR-V spec. Also note, that https://llvm.org/doxygen/classllvm_1_1DIStringType.html has encoding. If we can't reuse it now, I'd suggest to remove
```
  DIType *BaseTy =
      transDebugInst<DIType>(BM->get<SPIRVExtInst>(Ops[BaseTypeIdx])); 
```
entirely from the current version of the implementation. And anyway, it should be DIBasicType and not DIType.

Reply:
We can also preserve encoding by not using builder, but by directly calling [DIStringType::get](https://llvm.org/doxygen/classllvm_1_1MDNode.html#a7d10a7b9b7f40b04d27ed97c38ea1950)

Reply:
It's a bit trickier than using special Builder functions, but I agree - it'll allow easier support in future (in case if the signature changes).  Applied the suggestion.